lightning/src/ln/chan_utils.rs

```diff
@@ -482,75 +482,89 @@ impl_writeable_tlv_based!(HTLCOutputInCommitment, {
 	let payment_hash160 = Ripemd160::hash(&htlc.payment_hash.0[..]).into_inner();
 	if htlc.offered {
-		Builder::new().push_opcode(opcodes::all::OP_DUP)
-		              .push_opcode(opcodes::all::OP_HASH160)
-		              .push_slice(&PubkeyHash::hash(&revocation_key.serialize())[..])
-		              .push_opcode(opcodes::all::OP_EQUAL)
-		              .push_opcode(opcodes::all::OP_IF)
-		              .push_opcode(opcodes::all::OP_CHECKSIG)
-		              .push_opcode(opcodes::all::OP_ELSE)
-		              .push_slice(&countersignatory_htlc_key.serialize()[..])
-		              .push_opcode(opcodes::all::OP_SWAP)
-		              .push_opcode(opcodes::all::OP_SIZE)
-		              .push_int(32)
-		              .push_opcode(opcodes::all::OP_EQUAL)
-		              .push_opcode(opcodes::all::OP_NOTIF)
-		              .push_opcode(opcodes::all::OP_DROP)
-		              .push_int(2)
-		              .push_opcode(opcodes::all::OP_SWAP)
-		              .push_slice(&broadcaster_htlc_key.serialize()[..])
-		              .push_int(2)
-		              .push_opcode(opcodes::all::OP_CHECKMULTISIG)
-		              .push_opcode(opcodes::all::OP_ELSE)
-		              .push_opcode(opcodes::all::OP_HASH160)
-		              .push_slice(&payment_hash160)
-		              .push_opcode(opcodes::all::OP_EQUALVERIFY)
-		              .push_opcode(opcodes::all::OP_CHECKSIG)
-		              .push_opcode(opcodes::all::OP_ENDIF)
-		              .push_opcode(opcodes::all::OP_ENDIF)
-		              .into_script()
+		let mut bldr = Builder::new().push_opcode(opcodes::all::OP_DUP)
+			      .push_opcode(opcodes::all::OP_HASH160)
+			      .push_slice(&PubkeyHash::hash(&revocation_key.serialize())[..])
+			      .push_opcode(opcodes::all::OP_EQUAL)
+			      .push_opcode(opcodes::all::OP_IF)
+			      .push_opcode(opcodes::all::OP_CHECKSIG)
+			      .push_opcode(opcodes::all::OP_ELSE)
+			      .push_slice(&countersignatory_htlc_key.serialize()[..])
+			      .push_opcode(opcodes::all::OP_SWAP)
+			      .push_opcode(opcodes::all::OP_SIZE)
+			      .push_int(32)
+			      .push_opcode(opcodes::all::OP_EQUAL)
+			      .push_opcode(opcodes::all::OP_NOTIF)
+			      .push_opcode(opcodes::all::OP_DROP)
+			      .push_int(2)
+			      .push_opcode(opcodes::all::OP_SWAP)
+			      .push_slice(&broadcaster_htlc_key.serialize()[..])
+			      .push_int(2)
+			      .push_opcode(opcodes::all::OP_CHECKMULTISIG)
+			      .push_opcode(opcodes::all::OP_ELSE)
+			      .push_opcode(opcodes::all::OP_HASH160)
+			      .push_slice(&payment_hash160)
+			      .push_opcode(opcodes::all::OP_EQUALVERIFY)
+			      .push_opcode(opcodes::all::OP_CHECKSIG)
+			      .push_opcode(opcodes::all::OP_ENDIF);
+		if opt_anchors {		 
+			bldr = bldr
+				       .push_opcode(opcodes::all::OP_PUSHNUM_1)
```

Comment:
let's join this line with previous, and align the same as the previous block.  also below

Reply:
done